segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/DefragmentOperation.java

```diff
@@ -280,7 +280,7 @@
                 () -> {
                     readAtOffset.set(0);
                     val arg = concatArgs[currentArgIndex.get()];
-                    bytesToRead.set(Math.toIntExact(arg.getLength()));
+                    bytesToRead.set(arg.getLength());
```

Comment:
Why would we ever want to read more than 2GB at once? We don't have enough memory to store all that data.

Reply:
No we don't. 
It is the minimum of MaxBufferSizeForChunkDataTransfer and size. 


                    val buffer = new byte[Math.toIntExact(Math.min(chunkedSegmentStorage.getConfig().getMaxBufferSizeForChunkDataTransfer(), bytesToRead.get()))];

Reply:
So why is `bytesToRead` an `AtomicLong` ? What was wrong with `AtomicInt`?

Reply:
The chunk has long length and it may happen it actually is beyond Integer.Max
Casting it to/trying to fit into Int will cause overflow.
Note this is called from concatUsingAppend method.
In case of very long chunk we'll have to copy bytes from it in multiple batches.
After each read we update `bytesToRead` and `readAtOffset `
```
       bytesToRead.addAndGet(-size);
       readAtOffset.addAndGet(size);
```
making bytesToRead AtomicLong is the correct and minimal fix.

Math.toIntExact(Math.min(long, long)) will take care of it because getMaxBufferSizeForChunkDataTransfer is int